posthog/warehouse/models/external_data_schema.py

```diff
@@ -309,6 +309,45 @@ def filter_postgres_incremental_fields(columns: list[tuple[str, str]]) -> list[t
     return results
 
 
+def get_postgres_row_count(
+    host: str, port: str, database: str, user: str, password: str, schema: str
+) -> dict[str, int]:
+    connection = psycopg2.connect(
+        host=host,
+        port=port,
+        dbname=database,
+        user=user,
+        password=password,
+        sslmode="prefer",
+        connect_timeout=5,
+        sslrootcert="/tmp/no.txt",
+        sslcert="/tmp/no.txt",
+        sslkey="/tmp/no.txt",
+    )
+
+    try:
+        with connection.cursor() as cursor:
+            cursor.execute(
+                "SELECT tablename as table_name FROM pg_tables WHERE schemaname = %(schema)s",
+                {"schema": schema},
+            )
+            tables = cursor.fetchall()
+
+            if not tables:
+                return {}
+            union = [
+                f"SELECT '{table[0]}' AS table_name, COUNT(*) AS row_count FROM {schema}.{table[0]}" for table in tables
```

Comment:
Is it possible to SQL inject here? Can we parameterize this query instead? https://www.psycopg.org/psycopg3/docs/basic/params.html

Reply:
I considered this, the injection would need to be in the table name which I don't think is possible unless there is some crazy escape sequence hack. I'd also like to be more confident, so I'm going to add some substitutions when creating the query expression. I think the substituted approach reads a bit better as well, so no complaints there either.